gulpfile.js

```diff
@@ -787,13 +824,13 @@ function compileJs(srcDir, srcFilename, destDir, options) {
       .pipe(lazywrite())
       .on('end', function() {
         appendToCompiledFile(srcFilename, destDir + '/' + destFilename);
-        logBuildStep('Compiled', srcFilename);
-      }));
+      })).then(() => {
+        endBuildStep('Compiled', srcFilename, startTime);
```

Comment:
It looks like [`browserify.bundle()`](https://github.com/substack/node-browserify#bbundlecb) returns a [readable stream](https://nodejs.org/api/stream.html#stream_readable_streams) which fires events including "end" and "error". The "readable" event may be worth trying out in a follow-up PR.